linker: add custom align size to reduce alignment memory wasting

wentongwu opened the pull request:
1) When enable CONFIG_CUSTOM_SECTION_ALIGN, it need less alignment memory
for image rom region. But it will consume one more MPU region. For example somehow
we can know the size of .text and .rodata section, 512k+4B, with current code and
CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT enabled, the size of .text
and .rodata section in RAM (suppose only RAM) will be 1M.
But if with CONFIG_CUSTOM_SECTION_ALIGN(for example 64k), the size of .text and 
.rodata section will be 512k+64k(MPU_SUB_REGION_SIZE), that saved memory. But that needs carefully configure MPU region and sub-regions(ARMv7-M) to cover this feature.

2) Add custom align size for code relocation to reduce alignment memory wasting.

Fixes: #17337

nashif reviewed (changes requested):
Please change the PR title to something meaningful, no numbers please.

wentongwu committed:
linker: add custom align size to reduce alignment memory wasting

when enable CONFIG_CUSTOM_SECTION_ALIGN, it need less alignment
memory for image rom region. But that needs carefully configure
MPU region and sub-regions(ARMv7-M) to cover this feature.

Fixes: #17337.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>

wentongwu committed:
scripts: gen_relocate_app.py: add custom align size for relocation

add custom align size for code relocation to reduce alignment memory
wasting.

Fixes: #17337.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>
